src/Console/OutputStyle.php

```diff
@@ -97,4 +99,22 @@ public function throwable(Throwable $error): void
             $error = $error->getPrevious();
         } while ($error !== null);
     }
+
+    private function metaHeader(Message $message): AggregateHeader|StreamHeader
+    {
+        try {
+            return $message->header(AggregateHeader::class);
+        } catch (HeaderNotFound) {
+            return $message->header(StreamHeader::class);
+        }
+    }
+
+    private function streamName(AggregateHeader|StreamHeader $header): string
+    {
+        if ($header instanceof AggregateHeader) {
+            return StreamNameTranslator::streamName($header->aggregateName, $header->aggregateId);
+        }
+
+        return $header->streamName;
+    }
```

Comment:
maybe this should go into the `StreamNameTranslator` itself?  Since this is done in another places in a different style

Reply:
Then we have an dependency `aggregate->message`.

Currently, the aggregate knows nothing about messages. (Except for the AggregateHeader, which will be removed in the long term)

In 4.0, StreamNameTranslator can also be removed again; it is mainly needed as a BC layer.

Reply:
You mean domain wise? I mean, the Translator could also be moved into `Store`, which is imho even a better place, no? 

I know that this translator acts as a BC layer, if the remove the "old" store in favor of the new `StreamStore`. I'm fine keeping it as is right now, but if we encounter a third place, i would vote for getting this more centralized.

Reply:
How the stream name is composed is the task of the Aggregate/Repository domain. The store should not know anything about aggregate. In the future, only the repository will create this string.

I don't know if the Aggregate namespace is the right place, perhaps repository. But in my opinion, definitely not store.